_test/tests/inc/sort_with_collator.test.php

```diff
@@ -5,152 +5,220 @@
      */
-    public function test_intl_extension() {
+    public function testIntlExtenstionAvailability()
+    {
         if (!class_exists('Collator')) {
             $this->markTestSkipped('Skipping all sort tests with collator, as they need "intl" extension');
         }
         $this->assertTrue(true); // avoid being marked as risky for having no assertion
     }
 
     /**
-     * Pairs for collator-based intl_strcmp().
-     * @return array
+     * @return  Generator|array
+     * @see testStrcmp
+     */
+    public function provideWordPairs()
+    {
+        static $pairs = [
+            // Esperanto
+            'eo' => [
+                // if sort by 1st letter fails, sort by 2nd letter gives the opposite result
+                ['celo', 'ĉapo'],
+                ['ĉokolado', 'dento'], // c ĉ d
+                ['glacio', 'ĝirafo'],
+                ['ĝojo', 'haro'],
+                ['horo', 'ĥameleono'],
+                ['ĥoro', 'iam'], // g ĝ h ĥ i
+                ['jes', 'ĵaŭdo'],
+                ['ĵurnalo', 'kapo'], // j ĵ k
+                ['seka', 'ŝako'],
+                ['ŝuo', 'tablo'], // s ŝ t
+                ['urso', 'ŭaŭ'],
+                ['ŭo', 'vino'], // u ŭ v
+
+                // natural sort
+                ['paĝo 2', 'paĝo 10'],
+                ['paĝo 51', 'paĝo 100'],
+            ],
+        ];
+
+        foreach ($pairs as $lc => $list) {
+            foreach ($list as $pair) {
+                yield [$lc, $pair[0], $pair[1]];
+            }
+        }
+    }
+
+    /**
+     * Provide a list of correctly sorted "words"
+     *
+     * @return Generator|array
+     * @see testSort
+     * @see testKSort
+     * @see testASort
+     * @see testASortFnUrl
+     * @see testASortFnSafe
+     * @see testASortFnUtf8
      */
-    public function pairs() {
-        return array(
-            // if sort by 1st letter fails, sort by 2nd letter gives the opposite result
-            array('celo',     'ĉapo'     ),
-            array('ĉokolado', 'dento'    ), // c ĉ d
-            array('glacio',   'ĝirafo'   ),
-            array('ĝojo',     'haro'     ),
-            array('horo',     'ĥameleono'),
-            array('ĥoro',     'iam'      ), // g ĝ h ĥ i
-            array('jes',      'ĵaŭdo'    ),
-            array('ĵurnalo',  'kapo'     ), // j ĵ k
-            array('seka',     'ŝako'     ),
-            array('ŝuo',      'tablo'    ), // s ŝ t
-            array('urso',     'ŭaŭ'      ),
-            array('ŭo',       'vino'     ), // u ŭ v
-
-            // natural sort
-            array('paĝo 2',   'paĝo 10'  ),
-            array('paĝo 51',  'paĝo 100' )
-        );
+    public function provideSortedCharList()
+    {
+        static $data = [
+            'eo' => 'a b ci ĉe d e f go ĝi ho ĥa i ju ĵe k l m n o p r se ŝa t us ŭo v z', // Esperanto
+            'de' => 'a ä b c d e f g h i j k l m n o ö p q r s ß t u ü v w x y z', // German
+        ];
```

Comment:
Please reinsert the comment

    // if sort by 1st letter fails, sort by 2nd letter gives the opposite result

and use the advice given in the German data. For instance, use `az äb` and so on. **This would double-check the collactor.**

Reply:
I tried `an äh b c d e f g h i j k l m n os öl p q r so ßa t um üb v w x y z` but the tests failed.
It seems that `a = ä`, `o = ö`, `u = ü` and `s > ß` in the German collation. Is it true?